javascript/app-framework/src/components/editors/TableBundleEditor/TableBundleEditor.hooks.ts

```diff
@@ -55,18 +62,32 @@ export function useColumnState(): [
 	return [selectedColumn, onColumnClick]
 }
 
-function useTableBundleWorkflow(table: TableBundle): Workflow | undefined {
+function useTableBundleWorkflow(resource: TableBundle): Workflow | undefined {
 	return useMemo(
 		() =>
-			table
+			resource
 				.getSourcesWithProfile(KnownProfile.Workflow)
 				.find((t) => !!t) as Workflow,
 
 		/* eslint-disable-next-line react-hooks/exhaustive-deps */
-		[table, table.sources],
+		[resource, resource.sources],
 	)
 }
 
+export function useCodebookFields(resource: TableBundle): Field[] | undefined {
+	const codebook = useMemo(() => {
+		const result = resource
+			.getSourcesWithProfile(KnownProfile.Codebook)
+			.find((t) => !!t) as Codebook | undefined
+		return result
+		/* eslint-disable-next-line react-hooks/exhaustive-deps */
+	}, [resource, resource.sources])
+	const fields$ = useMemo(() => {
```

Comment:
You don't really need the return - and since we're expecting a possible `undefined` then maybe `??` would be more appropriate? e.g.`useMemo(() => codebook?.fields$ ?? from([])`